components/match2/wikis/teamfortress/match_group_input_custom.lua

```diff
@@ -0,0 +1,369 @@
+	map = mapFunctions.getExtraData(map)
+	map = mapFunctions.getScoresAndWinner(map)
+
+	return map
+end
+
+---@param record table
+---@param timestamp integer
+function CustomMatchGroupInput.processOpponent(record, timestamp)
+	local opponent = Opponent.readOpponentArgs(record)
+		or Opponent.blank()
+
+	-- Convert byes to literals
+	if Opponent.isBye(opponent) then
+		opponent = {type = Opponent.literal, name = 'BYE'}
+	end
+
+	---@type number|string
+	local teamTemplateDate = timestamp
+	-- If date is default date, resolve using tournament dates instead
+	-- default date indicates that the match is missing a date
+	-- In order to get correct child team template, we will use an approximately date and not the default date
+	if teamTemplateDate == DateExt.defaultTimestamp then
+		teamTemplateDate = Variables.varDefaultMulti('tournament_enddate', 'tournament_startdate', NOW)
+	end
+
+	Opponent.resolve(opponent, teamTemplateDate, {syncPlayer = true})
+	MatchGroupInput.mergeRecordWithOpponent(record, opponent)
+end
+
+-- called from Module:Match/Subobjects
+---@param player table
+---@return table
+function CustomMatchGroupInput.processPlayer(player)
+	return player
+end
+
+---@param data table
+---@param indexedScores table[]
+---@return table
+---@return table[]
+function CustomMatchGroupInput.getResultTypeAndWinner(data, indexedScores)
+	-- Map or Match wasn't played, set not played
+	if
+		Table.includes(NP_STATUSES, data.finished) or
+		Table.includes(NP_STATUSES, data.winner)
+	then
+		data.resulttype = 'np'
+		data.finished = true
+	-- Map or Match is marked as finished.
+	-- Calculate and set winner, resulttype, placements and walkover (if applicable for the outcome)
+	elseif Logic.readBool(data.finished) then
+		if MatchGroupInput.isDraw(indexedScores) then
+			data.winner = 0
+			data.resulttype = 'draw'
+			MatchGroupInput.setPlacement(indexedScores, data.winner, 1, 1)
+		elseif MatchGroupInput.hasSpecialStatus(indexedScores) then
+			data.winner = MatchGroupInput.getDefaultWinner(indexedScores)
+			data.resulttype = 'default'
+			if MatchGroupInput.hasForfeit(indexedScores) then
+				data.walkover = 'ff'
+			elseif MatchGroupInput.hasDisqualified(indexedScores) then
+				data.walkover = 'dq'
+			elseif MatchGroupInput.hasDefaultWinLoss(indexedScores) then
+				data.walkover = 'l'
+			end
+			indexedScores = MatchGroupInput.setPlacement(indexedScores, data.winner, 1, 2)
+		else
+			local winner
+			indexedScores, winner = MatchGroupInput.setPlacement(indexedScores, data.winner, 1, 2)
+			data.winner = data.winner or winner
+		end
+	end
+
+	--set it as finished if we have a winner
+	if Logic.isNotEmpty(data.winner) then
+		data.finished = true
+	end
+
+	return data, indexedScores
+end
+
+---@param opponents table[]
+---@param winner integer?
+---@param specialType string?
+---@param finished boolean|string?
+---@return table[]
+---@return integer?
+function MatchGroupInput.setPlacement(opponents, winner, specialType, finished)
```

Comment:
DO NOT DO THIS!

you are overwriting a commons function directly here
if you need a custom use the custom
if you actually want to use commons then kick this entirely

Reply:
What should I do, after removing the whole function (down to **return opponents, winner**) this happens 
![image](https://github.com/Liquipedia/Lua-Modules/assets/88981446/41163b97-09dc-47e6-9ff8-05b0264dd0e0)